### PR TITLE
Stage 2: surface-based coverage targets, CI gates, and artifact generators

### PR DESCRIPTION
**Pull request by neuron7xLab**
### Motivation
- Replace a single decorative `fail_under` number with evidence-backed coverage tied to risk surfaces so coverage becomes a verifiable engineering fact. 
- Prevent overclaiming (e.g. “98% verified”) by requiring per-surface thresholds, machine-readable artifacts and CI gates for critical modules.

### Description
- Added `configs/quality/coverage_targets.toml` to declare global gates and per-surface thresholds, claim risk and rationales for `core`, `execution`, `backtest`, `src`, `analytics`, `ingestion`, and `risk` surfaces.
- Added coverage tooling under `tools/coverage/`: `summarize_coverage.py` to produce `reports/coverage/coverage_summary.json`, `module_coverage.json` and `priority_map_YYYYMMDD.md`, `validate_coverage_targets.py` to assert the target contract, and `check_diff_coverage.py` to enforce a diff-coverage gate for changed critical files.
- Added test scaffolds for the required surface work: `tests/execution/test_negative_paths.py` (negative-paths for execution) and `tests/backtest/test_edge_cases.py` (backtest edge cases).
- Updated CI in `.github/workflows/main-validation.yml` to write coverage reports into `reports/coverage/`, run the summarize/validate/diff-gate steps, and upload coverage artifacts; updated `docs/METRICS_CONTRACT.md` with a new surface-based coverage claim entry.

### Testing
- Ran `pytest tests/execution/test_negative_paths.py tests/backtest/test_edge_cases.py tests/test_coverage_threshold_sync.py -q`, which failed due to a global test environment issue (deprecation of `asyncio.iscoroutinefunction` surfaced from `conftest.py` on Python 3.14) rather than the new Stage-2 scripts themselves.
- Exercised CLI/help paths for the new tools with `python tools/coverage/summarize_coverage.py --help`, `python tools/coverage/validate_coverage_targets.py --help`, and `python tools/coverage/check_diff_coverage.py --help`, all of which returned successfully.
- Verified bytecode/pack compilation with `python -m compileall tools/coverage tests/execution/test_negative_paths.py tests/backtest/test_edge_cases.py`, which succeeded.

------
[Codex Task](https://chatgpt.com/codex/cloud/tasks/task_e_6a1083290d7c8324a2275cca87106789)

### CODE REVIEW

File: .github/workflows/main-validation.yml [hunk 1]
```diff
@@ -49,13 +49,46 @@ jobs:
...
+            --coverage reports/coverage/coverage.xml \
+            --targets configs/quality/coverage_targets.toml
+
+      - name: Enforce diff coverage gate
+        run: |
+          .venv/bin/python tools/coverage/check_diff_coverage.py \
+            --coverage reports/coverage/coverage.xml \
+            --targets configs/quality/coverage_targets.toml \
+            --base origin/main \
+            --head HEAD \
```
[the remark this answers is not in the document]
Comment (neuron7xLab) [reply .github/workflows/main-validation.yml]: Бот помиляється: він критикує логіку, не враховуючи, що в контексті `push` до `main` гілка вже містить нові коміти. Порівнювати `HEAD` з `origin/main` некоректно, бо вони ідентичні. Бот вимагає використання попереднього `SHA`, а не `origin/main`.
